.travis.yml

```diff
@@ -3,6 +3,9 @@ language: c
 cache:
   directories:
   - $HOME/.stack
+  - $HOME/.ghc
+  - $HOME/.cabal
+  - .stack-work
```

Comment:
I just cargo cult my CI, but it seems weird that every one of your builds takes 2.5h+.

Reply:
I think the reason the builds take so long is mostly just cache misses. Travis has some rules around whether or not it will use a cache at all, based on some fairly conservative rules that I don’t totally understand. Builds that do use the cache are pretty fast, though, such as [this one](https://travis-ci.org/lexi-lambda/freer-simple/builds/445210407), so I don’t think caching more will help much here.